class2.1/task3.cpp

```diff
@@ -0,0 +1,74 @@
+#include <iostream>
+#include <fstream>
+#include <map>
+#include <string>
+//I create history.txt file for saving data.
+using namespace std;
+
+map<string, int> userHistory;
+
+void loadHistory() {
+    ifstream file("history.txt");
+    string name;
+    int count;
+    while (file >> name >> count) {
+        userHistory[name] = count;
+    }
+    file.close();
+}
+
+void saveHistory() {
+    ofstream file("history.txt");
+    for (const auto& entry : userHistory) {
+        file << entry.first << " " << entry.second << endl;
+    }
+    file.close();
+}
+
+void greetUser(const string& name) {
+    if (userHistory.find(name) == userHistory.end()) {
```

Comment:
`userHistory[name]++;` will handle both cases as based on C++ standard if no element is present in map then it will be created and the value will be default initialised which is for integer types means 0 value;